src/query/expression/src/evaluator.rs

```diff
@@ -502,6 +505,85 @@ impl<'a> Evaluator<'a> {
                 }
                 other => unreachable!("source: {}", other),
             },
+            (DataType::Variant, DataType::Array(inner_dest_ty)) => {
+                let empty_vec = vec![jsonb::Value::Null];
+                match value {
+                    Value::Scalar(Scalar::Variant(x)) => {
+                        let array = jsonb::from_slice(&x).map_err(|e| {
+                            ErrorCode::BadArguments(format!(
+                                "Expect to be valid json, got err: {e:?}"
+                            ))
+                        })?;
+                        let array = array.as_array().unwrap_or(&empty_vec);
+                        let column = VariantType::create_column_from_variants(array.as_slice());
+
+                        let validity = validity.map(|validity| {
+                            Bitmap::new_constant(
+                                validity.unset_bits() != validity.len(),
+                                column.len(),
+                            )
+                        });
+
+                        let new_array = self
+                            .run_cast(
+                                span,
+                                &DataType::Variant,
+                                inner_dest_ty,
+                                Value::Column(Column::Variant(column)),
+                                validity,
+                                options,
+                            )?
+                            .into_column()
+                            .unwrap();
+                        Ok(Value::Scalar(Scalar::Array(new_array)))
+                    }
+                    Value::Column(Column::Variant(col)) => {
+                        let mut array_builder =
+                            ArrayType::<VariantType>::create_builder(col.len(), &[]);
+
+                        for x in col.iter() {
+                            let array = jsonb::from_slice(x).map_err(|e| {
```

Comment:
need check validity first, NULL value is an empty slice, which will case `jsonb::from_slice` return an error.